plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -1,23 +1,148 @@
+ * Client Library.
+ */
 public class LocalAppEngineServerBehaviour extends ServerBehaviourDelegate {
+  private LocalAppEngineStartListener localAppEngineStartListener;
+  private LocalAppEngineExitListener localAppEngineExitListener;
+  private AppEngineDevServer devServer;
+
+  public LocalAppEngineServerBehaviour () {
+    localAppEngineStartListener = new LocalAppEngineStartListener();
+    localAppEngineExitListener = new LocalAppEngineExitListener();
+  }
 
   @Override
   public void stop(boolean force) {
     setServerState(IServer.STATE_STOPPING);
-    // TODO stop server
+    // stop server
+    terminate();
     setServerState(IServer.STATE_STOPPED);    
   }
 
-  void setStarted() {
-    setServerState(IServer.STATE_STARTED);    
+  /**
+   * Convenience method allowing access to protected method in superclass.
+   */
+  @Override
+  protected IModuleResourceDelta[] getPublishedResourceDelta(IModule[] module) {
+    return super.getPublishedResourceDelta(module);
+  }
+
+  /**
+   * Convenience method allowing access to protected method in superclass.
+   */
+  @Override
+  protected IModuleResource[] getResources(IModule[] module) {
+    return super.getResources(module);
+  }
+
+  /**
+   * Returns runtime base directory. Uses temp directory.
+   */
+  public IPath getRuntimeBaseDirectory() {
+    return getTempDirectory(false);
   }
-  
-  void setStarting() {
-    setServerState(IServer.STATE_STARTING);    
+
+  /**
+   * @return the directory at which module will be published.
+   */
+  public IPath getModuleDeployDirectory(IModule module) {
+    return getRuntimeBaseDirectory().append(module.getName());
+  }
+
+  /**
+   * Convenience accessor to protected member in superclass.
+   */
+  public void setModulePublishState2(IModule[] module, int state) {
+    setModulePublishState(module, state);
   }
-  
+
+  void startDevServer(List<File> runnables, MessageConsoleStream stream) {
+    setServerState(IServer.STATE_STARTING);
+
+    LocalAppEngineOutputLineListener outputListener =
+        new LocalAppEngineOutputLineListener(stream);
+
+    // Create dev app server instance
+    CloudSdk cloudSdk = new CloudSdk.Builder()
+        .addStdOutLineListener(outputListener)
+        .addStdErrLineListener(outputListener)
+        .startListener(localAppEngineStartListener)
+        .exitListener(localAppEngineExitListener)
+        .build();
+    devServer = new CloudSdkAppEngineDevServer(cloudSdk);
+    DefaultRunConfiguration devServerRunConfiguration = new DefaultRunConfiguration();
+    devServerRunConfiguration.setAppYamls(runnables);
+
+    // FIXME: workaround bug when running on a Java8 JVM
+    // https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools/issues/181
+    devServerRunConfiguration.setJvmFlags(Arrays.asList("-Dappengine.user.timezone=UTC"));
+
+    // Run server
+    try {
+      devServer.run(devServerRunConfiguration);
+      //setServerState(IServer.STATE_STARTED);
```

Comment:
Why is this commented out? My understanding is that this is critical for eclipse to tell that we've started.